Enhance virtual scrolling description (#1)
DevExpress/devextreme-documentation: api-reference/10 UI Widgets/dxDataGrid/1 Configuration/scrolling/mode.md

Specifies the scrolling mode.

---
The following scrolling modes are available:

- **Standard**      
A user scrolls a single page only. Scrolling is available only if [all the page's rows](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/#pageSize') do not fit into the widget's height. In this mode, the [pager](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/pager '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/pager/') performs the main navigation and scrolling is auxiliary. If [paging](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/') is disabled, the widget loads all rows simultaneously which can reduce the widget's performance. In this case, we recommend that you use another scrolling mode.

- **Infinite**      

> Each next page is loaded once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually, from the first to the last page.

Reviewer: ```suggestion
Each next page is loaded once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually from the first to the last page.
```

    [note]Set the **grouping**.[allowCollapsing](/api-reference/10%20UI%20Widgets/dxDataGrid/1%20Configuration/grouping/allowCollapsing.md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/grouping/#allowCollapsing') option to **false** when using infinite scrolling in conjunction with grouping.
    
- **Virtual**       

> Rows are loaded when they get into the viewport and removed once they leave it. Rows display gray boxes while they are being loaded and rendered. [Rendering optimization](/Documentation/Guide/Widgets/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization) may help reduce rendering time and remove the gray boxes. In this mode, users can move to any page instantly.

Reviewer: ```suggestion
Rows are loaded when they get into the viewport and removed once they leave. Rows display gray boxes while they are being loaded and rendered. [Rendering optimization](/Documentation/Guide/Widgets/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization) can reduce rendering time and remove gray boxes. In this mode, users can move to any page instantly.
```

> The following scrolling modes are available in the widget:

Reviewer: ```suggestion
The following scrolling modes are available:
```

- **Infinite**      
Each next page is loaded once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually, from the first to the last page.

    [note]Set the **grouping**.[allowCollapsing](/api-reference/10%20UI%20Widgets/dxDataGrid/1%20Configuration/grouping/allowCollapsing.md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/grouping/#allowCollapsing') option to **false** when using infinite scrolling in conjunction with grouping.
    
- **Virtual**       

> Rows are loaded when they get into the viewport and removed once they leave it. If the rows take time to be loaded and rendered, they display gray boxes. [Rendering optimization](/Documentation/Guide/Widgets/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization) may help reduce rendering time and remove the gray boxes. In this mode, users can move to any page instantly.

Reviewer: "If the rows take time" How would you quantify 'take time'?

Author: This depends on many things: local or remote data store, the presence of cell templates, the client's device. On old and slow devices it takes more time to render the widget.

Reviewer 2: I guess there is a value that rules whether the gray boxes should be displayed, but we shouldn't mention it in the docs because it can be changed, and the docs will be outdated.

Author: How would you say?

Reviewer: I would probably say something like "Row display gray boxes while they are loaded and rendered." This way it does not matter about how long it takes or which value determines it.

Reviewer 3: If rendering is fast the DataGrid does not show gray boxes. The DataGrid has internal criterios to determine fast and slow rendering. So, "while they are loaded and rendered" is not correct.

> A user scrolls a single page only. Scrolling appears only if [all the page's rows](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/#pageSize') do not fit into the widget's height. In this mode, the [pager](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/pager '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/pager/') performs the main navigation and scrolling is auxiliary. If [paging](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/') is disabled, the widget loads all rows at once, which may decrease the widget's performance. In this case, we recommend choosing another scrolling mode.

Reviewer: ```suggestion
A user scrolls a single page only. Scrolling is available only if [all the page's rows](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/#pageSize') do not fit into the widget's height. In this mode, the [pager](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/pager '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/pager/') performs the main navigation and scrolling is auxiliary. If [paging](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/') is disabled, the widget loads all rows simultaneously which can reduce the widget's performance. In this case, we recommend that you use another scrolling mode.
```

- **Virtual**       

> This mode is an alternative to paging where rows are loaded when they get into the viewport and removed once they leave it. While being loaded and rendered, the rows display gray boxes instead of data due to [Rendering Optimization](/Documentation/Guide/Widgets/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization). Use this mode if a user should be able to scroll data by pages.

Reviewer: ```suggestion
This mode is an alternative to paging where rows are loaded when they get into the viewport and removed once they leave it. Rows display gray boxes instead of data are when they are loaded and rendered due to [Rendering Optimization](/Documentation/Guide/Widgets/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization). Use this mode to allow a user to scroll through pages.
```

> This mode is an alternative to paging where pages are loaded when they get into the viewport and removed once they leave it. Use this mode if a user should be able to scroll data by pages. The data can appear as gray boxes while loading to avoid freezes and provide smooth scrolling. To optimize data rendering, refer to [Rendering Optimization](/Documentation/Guide/Widgets/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization) article.

Reviewer 2: ```suggestion
This mode is an alternative to paging where rows are loaded when they get into the viewport and removed once they leave it. While being loaded and rendered, the rows display gray boxes instead of data due to [Rendering Optimization](/Documentation/Guide/Widgets/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization). Use this mode if a user should be able to scroll data by pages.
```
